pallets/asset-manager/src/lib.rs

```diff
@@ -118,19 +123,26 @@ pub mod pallet {
 	#[pallet::generate_deposit(pub(crate) fn deposit_event)]
 	pub enum Event<T: Config> {
 		AssetRegistered(T::AssetId, T::AssetType, T::AssetRegistrarMetadata),
-		UnitsPerSecondChanged(T::AssetId, u128),
+		UnitsPerSecondChanged(T::AssetType, u128),
+		AssetTypeChanged(T::AssetId, T::AssetType),
 	}
 
-	/// Stores the asset TYPE
+	/// Stores the asset Type with assetId as key
```

Comment:
```suggestion
	/// Mapping from an asset id to asset type. This is mostly used when receiving transaction specifying an asset directly, like transferring an asset from this chain to another.
```
(We should provide a bit more explanation for our storage variable... I wrote an exemple but feel free to rewrite matching what correspond the best to it)